app/controllers/collection_numbers_controller.rb

```diff
@@ -200,7 +198,27 @@ def post_edit_collection_number
       @collection_number.destroy
       @collection_number = @other_number
     end
-    redirect_to_observation_or_collection_number
+
+    redirect_to_observation_or_object(@collection_number)
+  end
+
+  def check_for_form_errors?
```

Comment:
Super minor thing, but I thought the recommended convention for method names ending in "?" and "!" was:

? = just checks a condition and returns boolean, no side-effects

! = does whatever, like a normal method without the exclamation, but raises an error if something's wrong (think of the difference between ActiveRecord.create vs. create!, one silently fails, the other raises an error if it fails)

There are probably other additional cases, but these are the important ones here, because while, yes, this returns a boolean, it actually _does_ something if there are form errors, and sort of along the lines of raising an error, it redirects and that means all further processing should stop in this action.

So I would have expected this to be called "make_sure_there_are_no_form_errors!".  Or even "check_for_form_errors!" to keep the change minimal.

Another alternative would be to split this method up and call it like this:

reload_form && return if any_form_errors?

Wait, but again, it's not just checking for errors it's rendering an (appropriate) error message if there is one.  That should be made clear in the name.

reload_form && return if check_form_form_errors!

or

reload_form && return if flash_error_if_form_has_errors

or

flash_error_and_reload_if_form_has_errors

Yikes!  But at least it's now totally transparent what it does.  This is why the gods invented copy and paste. :)

Reply:
I never knew what those method punctuations really implied. I'll change these.

Reply:
Revision up. Let me know if that's not right

Reply:
Yes, so much better for me, thank you!